### PR TITLE
Fix circular includes Rtypes.h/TGenericClassInfo.h [ROOT-10820]

### PR DESCRIPTION
**Pull request by linev**
Move several class declarations into TGenericClassInfo.h,
include there only RtypesCore.h

### CODE REVIEW

**Comment by linev** [comment]
@pcanal Windows build is ok - just typical problem by uploading result back to Jenkins

**Comment by pcanal** [comment]
I am a bit confused.  What problem does this resolves?   The result seems sematically odd (many "types" are no longer defined in RTypes.h?

**Comment by linev** [comment]
Problem is in the name - Rtypes.h includes TGenericClassInfo.h. but TGenereicClassInfo.h includes Rtypes.h.  Current code only works if one include in the code Rtypes.h. If one tries to include TGenericClassInfo.h at first place - compilation fails

File: core/meta/inc/TGenericClassInfo.h [hunk 1]
```diff
@@ -12,19 +12,121 @@
 #ifndef ROOT_TGenericClassInfo
 #define ROOT_TGenericClassInfo
```
Comment (pcanal) [inline core/meta/inc/TGenericClassInfo.h:14]:
```suggestion
#ifndef ROOT_Rtypes
// Include RTypes.h outside of the code guard to insure the intended
// ordering of RTypes.h and TGenericClassInfo.h
#include "Rtypes.h"
#endif

#ifndef ROOT_TGenericClassInfo
#define ROOT_TGenericClassInfo
```
This also solves the problem while avoiding the need to migrate the type declaration from one file to the other.

File: core/meta/inc/TGenericClassInfo.h [hunk 2]
```diff
@@ -9,10 +9,13 @@
  * For the list of contributors see $ROOTSYS/README/CREDITS.             *
  *************************************************************************/
 
+#ifndef ROOT_Rtypes
+#include "Rtypes.h"
```
Comment (pcanal) [inline core/meta/inc/TGenericClassInfo.h:13]:
```suggestion
// Include RTypes.h outside of the code guard to insure the intended
// ordering of RTypes.h and TGenericClassInfo.h
#include "Rtypes.h"
```

**Comment by pcanal** [comment]
The multicore test failure is unrelated (and fixed by https://github.com/root-project/roottest/pull/535)

**Pull request merged**